app/code/community/Bolt/Boltpay/Helper/CatalogHelper.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+class Bolt_Boltpay_Helper_CatalogHelper extends Mage_Core_Helper_Abstract
+{
+    /**
+     * @return Mage_Core_Model_Abstract
+     */
+    public function getSession()
+    {
+        return Mage::getSingleton('catalog/session');
+    }
+
+    public function getLastRealOrderId()
+    {
+        return Mage::getSingleton('catalog/session')->getLastRealOrderId();
+    }
+
+    /**
+     * @param string $scope
+     * @return string
+     */
+    private function getQuoteIdKey($scope = 'ppc')
+    {
+        return $scope.'_quote_id_' . Mage::app()->getStore()->getId();
+    }
+
+    /**
+     * Get Quote for Product page
+     * @return Mage_Sales_Model_Quote
+     */
+    public  function getQuote()
+    {
+        $hasOrder = false;
+        $orderQuoteId = false;
+        $ppcQuoteId = $this->getSession()->getData($this->getQuoteIdKey());
+        if ($ppcQuoteId) {
+            $order = Mage::getModel('sales/order')->getCollection()->addAttributeToFilter('entity_id', $this->getLastRealOrderId())->getFirstItem()->setPageSize(1);
```

Comment:
It is better to move `setPageSize` before `getFirstItem`. Logically it is set page size for collection and `getFirstItem` retrieve collection first item
```
$order = Mage::getModel('sales/order')->getCollection()
->addAttributeToFilter('entity_id', $this->getLastRealOrderId())
->setPageSize(1)
->getFirstItem();
```